apiml-security-common/src/main/java/org/zowe/apiml/security/common/error/AuthExceptionHandler.java

```diff
@@ -38,143 +40,193 @@
 @Component
 public class AuthExceptionHandler extends AbstractExceptionHandler {
 
+    private final boolean isModulith;
+
     @SuppressWarnings("SpringJavaInjectionPointsAutowiringInspection")
     public AuthExceptionHandler(
         MessageService messageService,
-        ObjectMapper objectMapper) {
+        ObjectMapper objectMapper,
+        @Autowired(required = false) @Qualifier("isModulith") Boolean isModulith) {
         super(messageService, objectMapper);
+        this.isModulith = Boolean.TRUE.equals(isModulith);
+    }
+
+    @AllArgsConstructor
+    private static class HandlerContext {
+        String requestUri;
+        BiConsumer<ApiMessageView, HttpStatus> function;
+        BiConsumer<String, String> addHeader;
+    }
+
+    @FunctionalInterface
+    private interface ExceptionHandler {
+        void handle(RuntimeException ex, HandlerContext ctx);
+    }
+
+    private final Map<Class<? extends RuntimeException>, ExceptionHandler> exceptionHandlers = Map.ofEntries(
```

Comment:
```suggestion
    @FunctionalInterface
    private interface ExceptionHandler {
        <E extends Exception> void handle(E ex, HandlerContext ctx);
    }

    private <E extends Exception> Map.Entry<Class<? extends Exception>, ExceptionHandler> entry(Class<E> exceptionClazz, BiConsumer<E, HandlerContext> handler) {
        return Map.entry(exceptionClazz, (ExceptionHandler) handler);
    }

    private final Map<Class<? extends Exception>, ExceptionHandler> exceptionHandlers = Map.ofEntries(
```

This would simplify using a specific type in the handlers. Having RuntimeException everywhere is misleading and not useful.

Reply:
Actually this change breaks a lot of tests, I'll come back to this but I consider it lower priority

Reply:
I don't think there's much value in this change, let's change it later on